docs/docs/getting_started/tooling/noir_codegen.md

```diff
@@ -0,0 +1,135 @@
+---
+title: Noir Codegen for Typescript
+description: Learn how to use Noir codegen to generate Typescript bindings 
+keywords: [Nargo, Noir, compile, TypeScript]
+sidebar_position: 2
+---
+
+You can generate TypeScript bindings with `noir_codegen`, allowing for easier and faster integration of your Noir circuits into TypeScript projects.
+
+It is used in conjunction with `nargo export` which generates JSON for specified functions in a Noir library (not binary or contract circuits).
+
+## Installation
+
+You can install `noir_codegen` into your project using `yarn` or `npm`:
+
+```bash
+yarn add @noir-lang/noir_codegen
+```
+
+or 
+
+```bash
+npm install @noir-lang/noir_codegen
+```
+## Usage
+
+### Export ABI of specified functions
+
+Use the `#[export]` macro to indicate which functions in the noir library that you'd like to export.
+```rust
+#[export]
+fn your_function(...
+```
+
+The following command, when run from the directory with `Nargo.toml`, will create a folder with a .json file per exported function:
```

Comment:
```suggestion
At the root of your project, run:

\```
nargo export
\```

This will create a folder "exports" with a .json file per exported function.
```